### PR TITLE
[wip] await step over

### PR DESCRIPTION
**Pull request by jbhoosreddy**
Associated Issue: #3106

### Summary of Changes

* adds stepping over async call support

### Test Plan

- [x] Visual Testing

### CODE REVIEW

File: src/actions/breakpoints.js [hunk 1]
```diff
@@ -222,7 +229,7 @@ export function setBreakpointCondition(
   return ({ dispatch, getState, client, sourceMaps }: ThunkArgs) => {
     const bp = getBreakpoint(getState(), location);
     if (!bp) {
-      return dispatch(addBreakpoint(location, { condition }));
+      return dispatch(addBreakpoint(location, { condition, hidden: false }));
```
Comment (jasonLaster) [inline src/actions/breakpoints.js]: `hidden: false` should be the default: 

`dispatch(addBreakpoint(location, { condition }))`

File: src/actions/pause.js [hunk 2]
```diff
@@ -117,7 +120,7 @@ export function command({ type }: CommandType) {
 export function stepIn() {
   return ({ dispatch, getState }: ThunkArgs) => {
     if (getPause(getState())) {
-      return dispatch(command({ type: "stepIn" }));
+      return dispatch(AstCommand("stepIn"));
```
Comment (jasonLaster) [inline src/actions/pause.js]: `astCommand`

File: src/reducers/breakpoints.js [hunk 3]
```diff
@@ -384,4 +385,23 @@ export function getBreakpointsForSource(state: OuterState, sourceId: string) {
...
+
+export const getHiddenBreakpoint = createSelector(
+  getHiddenBreakpoints,
+  hiddenBreakpoints => {
+    if (hiddenBreakpoints.length) {
+      return hiddenBreakpoints[0];
+    }
+    return null;
+  }
+);
```
Comment (jasonLaster) [inline src/reducers/breakpoints.js]: maybe we're thinking about this the wrong way. If we're only going to have one temporary breakpoint, perhaps we want a different breakpoint field in the reducer like `pendingBreakpoints`.

I propose we add a `steppingBreakpoint` which does not appear in the UI... do you think that would be easy? we also don't need a pending breakpoint for this case.

File: src/utils/parser/steps.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    return;
+  }
+  for (let i = 0; i !== children.length; i++) {
+    const child = children[i];
+    if (containsPosition(child.loc, position)) {
+      const nextChild = children[++i];
+      const nextLocation = nextChild.loc.start;
+      nextLocation.sourceId = position.sourceId;
+      nextLocation.column = 0;
+      return nextLocation;
```
Comment (jasonLaster) [inline src/utils/parser/steps.js]: nit:

```js
return {
   sourceId: position.sourceId
   line: nextLocation.line,
   column: 0
}
```

File: src/utils/parser/steps.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { Source } from "../../../flow-typed/debugger-html";
+import { AstPosition } from "./types";
+import { getClosestPath } from "./utils/closest";
+import { isAwaitExpression, containsPosition } from "./utils/helpers";
+import type { NodePath } from "babel-traverse";
+
+export function getNextStep(
```
Comment (jasonLaster) [inline src/utils/parser/steps.js]: it would be nice to have unit tests here too.

File: src/actions/breakpoints.js (hunk 1 above)
Comment (jbhoosreddy) [inline src/actions/breakpoints.js]: flow complained here. maybe there's a better solution.

File: src/actions/breakpoints.js [hunk 6]
```diff
@@ -95,15 +95,25 @@ export function syncBreakpoint(
...
   return ({ dispatch, getState, sourceMaps, client }: ThunkArgs) => {
     const action = { type: "ADD_BREAKPOINT", breakpoint };
     const promise = addBreakpointPromise(getState, client, sourceMaps, action);
     return dispatch({ ...action, [PROMISE]: promise });
   };
 }
 
+export function addHiddenBreakpoint(location: Location) {
+  return ({ dispatch }: ThunkArgs) => {
+    return dispatch(addBreakpoint(location, "", true));
```
Comment (jasonLaster) [inline src/actions/breakpoints.js]: at this point it would make sense to do `addBreakpoint({location, condition, hidden})`

**Pull request closed without merging**